### PR TITLE
General improvement part3: separate NN structure for graph and node level output

### PR DESCRIPTION
**Pull request by pzhanggit**
1. Separate NN structures for graph and node level predictions
2. Make NN structures for graph and node configurable
3. Tested for node level prediction locally, though not in CI test yet

### CODE REVIEW

File: test_trained_model.py
Comment (streeve) [inline test_trained_model.py:44]: Why do we need to add to the config?

File: models/GATStack.py
Comment (streeve) [inline models/GATStack.py:16]: I assume this is for consistency across models

File: examples/configuration.json
Comment (streeve) [inline examples/configuration.json:24]: I think I need you to explain this to me in more detail

**Review by streeve — COMMENTED** [review]
Can you post accuracy details from local runs? I'm curious how much this improves results. I also think I need a better understanding before I can approve

File: test_trained_model.py
Comment (pzhanggit) [reply test_trained_model.py:44]: Because different NN structures are designed for graph and node outputs.

**Comment by pzhanggit** [comment]
@streeve See f6c97c3 for new changes we discussed earlier.

**Review by streeve — COMMENTED** [review]
Looks great - I'll approve if you add a comment that you tested with node-level outputs (not in the CI yet) and fix the two really minor things below

File: models/Base.py
Comment (pzhanggit) [reply models/Base.py:81]: Yeah, agreed, I will change it.  I think neither of them should be a member variable unless it shows big improvement in performance.

File: examples/ci.json
Comment (pzhanggit) [reply examples/ci.json:5]: I would keep it. It comes together with dataset.

**Comment by pzhanggit** [comment]
> Looks great - I'll approve if you add a comment that you tested with node-level outputs (not in the CI yet) and fix the two really minor things below

See f9a5b34 and also the update description.

File: examples/ci.json
Comment (streeve) [reply examples/ci.json:5]: I didn't realize this has no effect on the unit test - problem for another PR

**Review by streeve — APPROVED** [review]

**Pull request merged**
